core/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -257,7 +257,7 @@ private[spark] object Logging {
         var logger = LogManager.getLogger(logEvent.getLoggerName)
           .asInstanceOf[org.apache.logging.log4j.core.Logger]
         while (logger.getParent() != null) {
```

Comment:
Does it need to check all parents? If config of "this" logger is equal to root config, it seems to me that the config of its parents are equal to root too.

Reply:
Oh, yea, this is different to log4j 1.x.